src/components/application_manager/src/resumption/resumption_data_processor_impl.cc

```diff
@@ -79,40 +79,54 @@ void ResumptionDataProcessorImpl::Restore(
     ResumeCtrl::ResumptionCallBack callback) {
   SDL_LOG_AUTO_TRACE();
 
-  if (!HasDataToRestore(saved_app)) {
+  bool restore_hmi_data = HasHMIDataToRestore(saved_app);
+  bool restore_local_data = HasLocalDataToRestore(saved_app);
+  if (!restore_hmi_data && !restore_local_data) {
     SDL_LOG_DEBUG("No data to restore, resumption is successful");
     callback(mobile_apis::Result::SUCCESS, "Data resumption succesful");
     return;
   }
 
+  const auto app_id = application->app_id();
+  if (!restore_hmi_data && restore_local_data) {
+    AddSubscriptions(application, saved_app);
+    SDL_LOG_DEBUG("No requests to HMI for " << app_id
+                                            << ", resumption is successful");
+    callback(mobile_apis::Result::SUCCESS, "Data resumption successful");
+    return;
+  }
+
```

Comment:
@ychernysheva I believe you can remove this block and just add in the end of function smth like:
```
if (!restore_hmi_data && restore_local_data) {
  SDL_LOG_DEBUG("No requests to HMI for " << app_id
                                            << ", resumption is successful");
  callback(mobile_apis::Result::SUCCESS, "Data resumption successful");  
}
```

The rest of functions will just do nothing if there is no HMI data to resume

Reply:
@AKalinich-Luxoft I think that we need  this code, because in fact all non-HMI subscriptions are resumed in method `AddSubscriptions(application, saved_app)` from this code block, if we don't have any HMI subscriptions and don't need to care about their responses from HMI.

Now such non-HMI subscriptions include AppServices and SystemCapabilities (now don't work and need to be fixed in the future, but we still need to ensure that we have place to process them even they don't work now).

Also there is a possibility that in the future some other type of non-HMI subscriptions will be added, which also will be processed by plugins or with other means, so in this case we will need to simply add corresponding calls to this code block.

Reply:
@AKalinich-Luxoft Also some clarifications are added in f923c29089b198c9dff73c07a1f705164eade5b3